src/main.cpp

```diff
@@ -794,7 +796,7 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
 
     // Rather not work on nonstandard transactions (unless -testnet/-regtest)
     string reason;
-    if (fRequireStandard && !IsStandardTx(tx, reason))
+    if (!policy.ApproveTx(tx, reason))
```

Comment:
I almost agree for libconsensus (soft/hardforks may require a change in the API though).
For policy changes are more likely to be expected (ie someone asks for a change in the API so that it's possible to support some new policy).